archunit/src/main/java/com/tngtech/archunit/library/modules/syntax/ModulesByAnnotationShould.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2014-2023 TNG Technology Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tngtech.archunit.library.modules.syntax;
+
+import java.lang.annotation.Annotation;
+
+import com.tngtech.archunit.PublicAPI;
+import com.tngtech.archunit.base.DescribedPredicate;
+import com.tngtech.archunit.core.domain.Dependency;
+import com.tngtech.archunit.core.domain.JavaClasses;
+import com.tngtech.archunit.core.domain.PackageMatcher;
+import com.tngtech.archunit.lang.ArchRule;
+import com.tngtech.archunit.library.modules.AnnotationDescriptor;
+import com.tngtech.archunit.library.modules.ArchModule;
+
+import static com.tngtech.archunit.PublicAPI.Usage.ACCESS;
+
+@PublicAPI(usage = ACCESS)
+public interface ModulesByAnnotationShould<ANNOTATION extends Annotation> extends ModulesShould<AnnotationDescriptor<ANNOTATION>> {
+
+    /**
+     * Like {@link #respectTheirAllowedDependencies(DescribedPredicate, ModuleDependencyScope)}, but the allowed dependencies will be automatically
+     * derived from the {@link ANNOTATION} property named {@code annotationPropertyName}. This property *must* be of type {@code String[]}
```

Comment:
Why is `ANNOTATION` a `@link`? (I would have expected `@code`. Isn't it a parameter just like `annotationPropertyName`?)

Reply:
The reason for me was that `annotationPropertyName` is a locally scoped thing (and I can't even link to it if I want), but the scope of `ANNOTATION` is wider :man_shrugging: Making it a link makes it immediately clear for me, that I mean the class type parameter :man_shrugging: